Algorithm.Python/EmaCrossUniverseSelectionAlgorithm.py

```diff
@@ -32,76 +32,86 @@
-        # - coarse selection function: accepts an IEnumerable<CoarseFundamental> and returns an IEnumerable<Symbol>
         self.AddUniverse(self.CoarseSelectionFunction)
-
-
-    # sort the data by daily dollar volume and take the top 'NumberOfSymbols'
-    def CoarseSelectionFunction(self, coarse):
-
-        # We are going to use a dictionary to refer the object that will keep the moving averages
-        for cf in coarse:
-            if cf.Symbol not in self.averages:
-                self.averages[cf.Symbol] = SymbolData(cf.Symbol)
-
-            # Updates the SymbolData object with current EOD price
-            avg = self.averages[cf.Symbol]
-            avg.update(cf.EndTime, cf.AdjustedPrice)
-
-        # Filter the values of the dict: we only want up-trending securities
-        values = list(filter(lambda x: x.is_uptrend, self.averages.values()))
-
-        # Sorts the values of the dict: we want those with greater difference between the moving averages
-        values.sort(key=lambda x: x.scale, reverse=True)
-
-        for x in values[:self.coarse_count]:
-            self.Log('symbol: ' + str(x.symbol.Value) + '  scale: ' + str(x.scale))
-
-        # we need to return only the symbol objects
-        return [ x.symbol for x in values[:self.coarse_count] ]
-
-    # this event fires whenever we have changes to our universe
+        self.averages = {}
+        
+    def CoarseSelectionFunction(self,coarse):
+        
+        sorted_by_volume = sorted([cf for cf in coarse if cf.Price > 10],
+            key = lambda cf: cf.DollarVolume, reverse=True)
+            
+        selected = {cf.Symbol: cf for cf in sorted_by_volume[:100]}
+        symbols = list(selected.keys())
+        
+        new_symbols = [s for s in symbols if s not in self.averages]
+        if new_symbols:
+            
+            # Get history for all new symbols. If empty, log and return the selected -- WOW!
+            history = self.History(new_symbols, 200, Resolution.Daily)
+            if history.empty:
+                self.Debug(f'Empty history on {self.Time} for {new_symbols}')
+                return Universe.Unchanged   # Continue with previous universe
+                
+            # Only need the closing prices    
+            history = history.close.unstack(0)
+            
+            # Add new item to self.averages
+            for symbol in new_symbols:
+                if symbol in history:
+                    self.averages[symbol] = SelectionData(history[symbol].dropna())
+                    
+        self.symbols = []
+        
+        for symbol, cf in selected.items():
+            symbolData = self.averages[symbol]
+            
+            if symbol not in new_symbols:
+                symbolData.Update(cf.EndTime, cf.AdjustedPrice) 
+                
+            if symbolData.IsReady and symbolData.fast > symbolData.slow:
+                self.symbols.append(symbol)
+                
+        return self.symbols
```

Comment:
In the former algorithm, we choose the symbols with the highest scale
```python
self.scale = (fast - slow) / ((fast + slow) / 2.0)
```
also there is a tolerance in the fast > slow condition:
```
if symbolData.IsReady and symbolData.fast > symbolData.slow * self.tolerance:
    self.symbols.append(symbol)
```